core/src/main/kotlin/plugability/DokkaPlugin.kt

```diff
@@ -9,6 +9,11 @@ import kotlin.properties.ReadOnlyProperty
 import kotlin.reflect.KProperty
 import kotlin.reflect.KProperty1
 
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING, message = "The Plugin API is experimental. It can be incompatibly changed in the future.")
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
+@Retention(AnnotationRetention.BINARY)
+annotation class ExperimentalDokkaPluginApi
```

Comment:
I wonder if it should be named `PluginApiPreview` instead (or something similar with `Preview`). Two reasons:

`ExperimentalDokkaPluginApi` sounds too generic, like something that should be used selectively and for features/api with smaller scope. If we introduce it now with this name and make users suppress it on plugin/file/project level, they will probably never remove it as long as there's some experimental API and the annotation is "used". However, they won't know that some API became stable and some other API (including new) became experimental.

With `Preview`, we're clearly grouping a set of API (all of it in our case) under one annotation. After we revise plugin API, we can remove the `Preview` annotation and the user should hopefully notice it (useless `OptIn` annotation inspection).

However, I am almost certain that after we revise the API, not all of it will become stable - this is where `ExperimentalDokkaPluginApi` would come in handy. We'd make, for example, 90% of API stable, and then mark remaining 10% with `ExperimentalDokkaPluginApi` (or something similar, maybe also under a group if possible).

___

`Experimental` sounds scary, as if we can remove it altogether without providing a migration path or even an alternative. I would be scared to build plugins around that. `Preview` indicates to me that some thought was put into designing the API, but it's not final yet and it might change, but not to the point where your plugin cannot exist anymore. The latter seems to be our case - we'll most likely just refactor it: name/package changes, data class to class migration, enum to sealed object, move extension points around, etc. All of these changes, even though breaking, are manageable with proper migration guides. So `Preview` to me better describes awaiting changes